src/utils.rs

```diff
@@ -47,6 +47,33 @@ pub fn from_relocatable_to_indexes(relocatable: Relocatable) -> (usize, usize) {
         (relocatable.segment_index as usize, relocatable.offset)
     }
 }
+// ================
+//   RunResources
+// ================
+
+/// Maintains the resources of a cairo run. Can be used across multiple runners.
+#[allow(dead_code)]
+pub struct RunResources {
+    n_steps: Option<usize>,
```

Comment:
`RunResources` is already an optional argument, do we need an option inside it too?

Reply:
Using 0 as default would lead to unwanted behaviour. What I mean is that if the `RunResources` struct is itself optional and its inner value is never updated when it is set as  None, then all use cases in which the function would receive a run_resources = RunResources(None) can jus receive run_resources = None. In that case we don't need a default value